interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java

```diff
@@ -31,38 +33,78 @@
  * navigation      0.615594        http://411mania.com/games/dragon-fantasy-book-one-psn-review/   http://411mania.com/games/dragon-fantasy-book-one-psn-review/   Sovrn
  */
 public class AverageScore {
+
     public static void main(String[] args) throws Exception {
+        if(args.length < 2)
+            System.exit(2);
+        System.exit(mapReduce(args[0], args[1]));
+    }
+
+    public static int mapReduce(String inputPath, String outputPath) throws Exception {
         Configuration conf = new Configuration();
         Job job = Job.getInstance(conf, "Interview Averaging");
         job.setJarByClass(AverageScore.class);
         job.setMapperClass(AverageScoreMapper.class);
         job.setReducerClass(AverageScoreReducer.class);
-        // TODO: Finish the key and output setup
+        job.setOutputKeyClass(Text.class);
+        job.setOutputValueClass(FloatWritable.class);
 
-        FileInputFormat.addInputPath(job, new Path(args[0]));
-        FileOutputFormat.setOutputPath(job, new Path(args[1]));
+        FileInputFormat.addInputPath(job, new Path(inputPath));
+        FileOutputFormat.setOutputPath(job, new Path(outputPath));
 
-        System.exit(job.waitForCompletion(true) ? 0 : 1);
+        return job.waitForCompletion(true) ? 0 : 1;
     }
 
     public static class AverageScoreMapper extends Mapper<Object, Text, Text, FloatWritable> {
 
+        private final FloatWritable number = new FloatWritable();
+        private final Text domainText = new Text();
+
         @Override
         protected void map(Object key, Text value, Mapper<Object, Text, Text, FloatWritable>.Context context)
                 throws IOException, InterruptedException {
-            // TODO : Implement the mapping phase
+            String[] values = value.toString().split("\t");
             
+            if(values.length < 4)
+            	return;
+
+            try {
+                String domain = new URI(values[3]).getHost();
+                number.set(Float.parseFloat(values[1]));
+
+                if(domain != null) {
+                    domainText.set(domain);
+                    context.write(domainText, number);
+                }
+
+            } catch (URISyntaxException e) {
+                // Invalid data - not a clean URL. No error reporting requirement, so ignore for now.
+            } catch (NumberFormatException e) {
+                // Invalid data - value wasn't a number (as in the column header line). 
+            	// No error reporting requirement, so ignore for now.
```

Comment:
Didn't think designing an error-reporting component was a reasonably-sized addition to a test that wasn't supposed to be taking very long.